XiEditor/LineCache.swift

```diff
@@ -178,4 +199,32 @@ class LineCache {
         }
         return result
     }
+
+    /// Set of lines that need to be invalidated to blink the cursor
+    var cursorInval: InvalSet {
+        let inval = InvalSet()
+        for (i, line) in lines.enumerated() {
+            if line?.containsCursor ?? false {
+                inval.addRange(start: i + nInvalidBefore, n: 1)
+            }
+        }
+        return inval
+    }
+}
+
+/// A set of line numbers to be invalidated, in run-length representation
+class InvalSet {
+    var ranges: [(Int, Int)] = []
```

Comment:
Non-block: I'd be inclined to use either the `Range` or `ClosedRange` foundation type here.